### PR TITLE
Add FilesystemConnector with deterministic enumeration, refactor InMemory connector tests

### PR DESCRIPTION
**Pull request by ahrav**
## Summary

Add a Unix-only filesystem-backed `EnumerationConnector` + `ReadConnector` that
indexes regular files under a root directory and provides deterministic key-range
scans with binary search pagination and byte-weighted split-point selection.
Also refactors the in-memory connector with precomputed metadata and extracts
both connector test suites into standalone files with rstest parameterization.

**Impact**: 14 files changed (+3050, -784) — net +2266 lines
**Risk Level**: Medium (new feature + refactored existing connector internals)

### New Files

| File | Lines | Purpose |
|------|-------|---------|
| `crates/gossip-connectors/src/filesystem.rs` | 937 | `FilesystemConnector` implementation |
| `crates/gossip-connectors/src/filesystem_tests.rs` | 1138 | 50 test fns + rstest parameterized cases |
| `crates/gossip-connectors/src/in_memory_tests.rs` | 694 | Extracted in-memory tests (22 test fns + rstest) |

### Modified Files

| File | Change |
|------|--------|
| `crates/gossip-connectors/src/in_memory.rs` | Refactored: `PreparedItem` precomputation, `Arc`-shared internals, unique key requirement, improved resume semantics with token validation in all builds |
| `crates/gossip-connectors/src/lib.rs` | Added `filesystem` module + `FilesystemConnector` re-export |
| `crates/gossip-stdx/src/inline_vec.rs` | Kani fix: element-wise writes for `from_slice` verification, unwind bound bump |
| `crates/gossip-connectors/Cargo.toml` | Added `rstest` + `proptest` dev-dependencies |
| `docs/boundary-{1,2,5}*.md`, `docs/coordination-testing.md`, `docs/simulation-harness.md` | Updated 5 design docs to match current code |

### FilesystemConnector

- **Lazy one-shot indexing**: First enumerate/split call walks the directory tree using an explicit stack (no recursion). Entries are globally sorted by key bytes for deterministic pagination.
- **Symlink skipping**: `d_type`-based classification without following links. Prevents symlink-cycle hangs and root-escape via symlink targets.
- **Byte-weighted split points**: `choose_split_point` selects where cumulative file size crosses the halfway mark, balancing shards by byte volume rather than item count.
- **Path-based ItemRef**: Unlike the in-memory connector's positional index refs, filesystem refs are raw relative-path bytes — stable across re-indexing of the same snapshot.
- **Weak VersionId**: Metadata-derived (`mtime`, `mtime_nsec`, `len`, `inode`) to avoid content hashing during index build.
- **TOCTOU documented**: File index is built once; reads reopen by path. Version staleness is caller's responsibility at higher layers.

### InMemory Connector Refactor

- **PreparedItem**: Precomputes `StableItemId`, `ObjectVersionId`, `ItemRef`, and size hint at construction time — eliminates redundant BLAKE3 hashing on every page.
- **Token validation in all builds**: Token-vs-key cross-check now runs in release too (was debug-only). Stale/missing tokens fall back to O(log n) binary search.
- **Unique key panic**: Constructor panics on duplicate keys (previously silent data corruption). Documented why: cursor resume skips duplicates, identity collides, split-point produces empty shards.
- **Budget expiry returns retryable error** instead of empty page (avoids ambiguity with EOF signal).

### InlineVec Kani Fix

- Added `#[cfg(kani)]` element-wise write path for `from_slice` because Kani/CBMC cannot track symbolic values through `copy_nonoverlapping` into `MaybeUninit` pointer casts. Production retains the bulk memcpy (validated by Miri).

## Testing

**72 test functions** across the two new test files, plus rstest parameterization expanding to significantly more test cases:

### Filesystem Tests (50 `#[test]` + 1 `#[rstest]`)
- Conformance suite: empty connector, single item, page budget, split point
- Enumeration: directory trees, nested dirs, empty files, large trees (100+ files)
- Shard bounds: partial ranges, single-key ranges, disjoint ranges
- Token parity: token-enabled vs token-disabled produce identical results
- Split points: byte-weighted median, equal-size files, zero-size files
- Filesystem-specific: symlinks skipped, unreadable dirs, deep nesting (100 levels), non-UTF8 filenames, special characters, empty directories
- Property-based: deterministic enumeration across multiple calls

### In-Memory Tests (22 `#[test]` + 5 `#[rstest]` with 23 `#[case]`)
- Converted 28 original tests to rstest parameterization
- Conformance, shard bounds, split point, read_range, invalid ItemRef cases
- Helper `enumerate_single_item_ref()` reduces boilerplate

## Test Plan

- [x] `cargo fmt --all` passes
- [x] `cargo check` passes
- [x] `cargo clippy --all-targets --all-features -- -D warnings` passes
- [x] `cargo test -p gossip-connectors` — all 72+ tests pass
- [x] Kani verification: `from_slice_bounds_are_sound` passes with new unwind bound
- [ ] Full `cargo test --workspace` (reviewer should verify)
- [ ] Verify on macOS: deep directory test uses 100 levels with 2-char names to stay within `NAME_MAX`/`PATH_MAX` limits

## Risk Assessment

| Factor | Level | Notes |
|--------|-------|-------|
| Size | Medium | 3050 insertions, but ~1800 are new test code |
| Complexity | Medium | Filesystem walk + binary search pagination are well-documented |
| Backwards compat | Low | New connector is additive; InMemory refactor changes internal behavior (unique key panic, token validation) but no external API change |
| Test coverage | Low risk | 72+ test functions covering conformance, edge cases, properties |
| Platform | Medium | `#[cfg(unix)]` — requires Unix for `OsStrExt` byte access |

## Design Doc Updates

Five design docs updated to reflect current code state:
- `boundary-1-identity-spine.md`: Added `coordination.rs` to file inventory, bumped golden vector count
- `boundary-2-coordination.md`: `ShardSnapshot` → `ShardSnapshotView`, `mod.rs` → `lib.rs`
- `boundary-5-persistence.md`: Fixed CAS condition status codes to match actual enum discriminants
- `coordination-testing.md`: Updated test file declarations, added simulation test files
- `simulation-harness.md`: Minor updates

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/gossip-connectors/src/filesystem.rs [hunk 1]
```diff
@@ -0,0 +1,937 @@
...
+        start: Option<&ItemKey>,
+        end: Option<&ItemKey>,
+        cursor: &Cursor,
+        budgets: Budgets,
+    ) -> Result<EnumerationPage, EnumerateError> {
+        self.ensure_indexed()?;
+
+        if budgets.is_expired_at(Instant::now()) {
+            return Ok(EnumerationPage::new(Vec::new(), cursor.clone()));
+        }
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-connectors/src/filesystem.rs]: Same as the other comment — the code at this line already returns `Err(EnumerateError::retryable("budget deadline expired"))`, which is exactly the suggested fix. The test `expired_budget_returns_retryable_error` confirms this.

File: crates/gossip-connectors/src/filesystem.rs [hunk 2]
```diff
@@ -0,0 +1,937 @@
...
+        let split_idx = if total_bytes == 0 {
+            // All files empty; fall back to count-balanced.
+            start_idx + range.len() / 2
+        } else {
+            let half = total_bytes / 2;
+            let mut cumulative = 0u64;
+            // Default to count-median; overwritten once cumulative crosses half.
+            let mut idx = start_idx + range.len() / 2;
+            for (i, file) in range.iter().enumerate() {
+                cumulative += file.size;
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-connectors/src/filesystem.rs]: The cumulative loop already uses `saturating_add` (line 496 in current code). The `.sum()` on total_bytes (line 487) could theoretically wrap, but would require >18 exabytes of total file size in a single shard range — not a realistic scenario for a directory walker. The `saturating_add` on the incremental accumulator is the one that matters for correctness (it determines where the split lands), and that's already in place.

That said, switching `.sum()` to `.fold(0u64, u64::saturating_add)` for consistency with the loop is reasonable. Applied.

File: crates/gossip-connectors/src/in_memory.rs [hunk 3]
```diff
@@ -312,7 +421,29 @@ impl InMemoryDeterministicConnector {
...
         }
 
-        let split_idx = start_idx + (range_end - start_idx) / 2;
+        // Byte-weight median: find the item whose cumulative size first
+        // reaches half the total byte volume. This produces balanced shards
+        // even when object sizes are highly skewed.
+        let slice = &self.items[start_idx..range_end];
+        let total_bytes: u64 = slice.iter().map(|i| i.size_hint).sum();
+        let half = total_bytes / 2;
+        let mut cumulative = 0u64;
```
[the remark this answers is not in the document]
Comment (ahrav) [reply crates/gossip-connectors/src/in_memory.rs]: The in-memory connector's cumulative loop already uses `saturating_add` (line 433). Switched the `.sum()` on line 428 to `.fold(0u64, u64::saturating_add)` for consistency.
